### DISCUSSION TITLE
Multiple Filter OR condition not working

### DISCUSSION THREAD

**Discussion by azdave2020** [nextgenhealthcare/connect#5257] [General]
OK I am sure I am missing something.  The ask was simple, filer out HL7 messages that contain LOINC 69937-1 or LOINC 223222-2 in OBX3.1  It does not make an error, it just fails to filter either test message. Here are the shots of the setup
![image](https://user-images.githubusercontent.com/90641786/176030857-cc988cf6-3ec8-4dde-8187-d46517ec42b9.png)

![image](https://user-images.githubusercontent.com/90641786/176030922-ef8bcd34-38af-41a7-9695-e8d90d9e6180.png)

I am on Mirth 3.10.1
Java java version "1.8.0_331"
Java(TM) SE Runtime Environment (build 1.8.0_331-b09)
Java HotSpot(TM) Client VM (build 25.331-b09, mixed mode)

What am I missing?

**Comment by mjtrac** [comment]
Most likely you are receiving messages with multiple OBX segments.  You can either switch to an iterator (if you used drag and drop to build your filters, a dialog box probably popped up to offer) or write a javascript for loop as a filter.  If you use an iterator, you only need one rule under it, which can handle both values.

![image](https://user-images.githubusercontent.com/2815700/176039085-ebd70017-57f6-4f68-864b-82d0afe0b31a.png)

**Comment by azdave2020** [comment]
Testing it now will post final---thanks for the tips

**Comment by tonygermano** [comment]
You may have an issue as @mjtrac suggested if your messages contain multiple OBX segments.

You definitely have a logic issue even if there is only a single OBX segment. You want to use AND instead of OR. I think the negative predicates (does not contain) makes it a little confusing.

AND means both conditions must be true - **OBX-3.1 does not contain 69937-1 AND OBX-3.1 does not contain 223222-2**. If the value is 223222-2, then the second clause is false, so the whole thing is false and the message gets filtered.

OR means that either condition can be true - **OBX-3.1 does not contain 69937-1 OR OBX-3.1 does not contain 223222-2**. If the value is 223222-2, then the first clause is true, so the whole thing is true and the message gets accepted.
